pkgs/os-specific/linux/kernel-headers/default.nix

```diff
@@ -93,6 +93,9 @@ in {
       };
       patches = [
          ./no-relocs.patch # for building x86 kernel headers on non-ELF platforms
+      ] ++ lib.optionals (stdenvNoCC.buildPlatform.isDarwin &&
```

Comment:
I think we generally try to have patches not be conditional.  Your patch already has `#ifdef APPLE` so it should be safe to apply it unconditionally, right?